basics/getting_started/plugin_compatibility.md

```diff
@@ -2,80 +2,102 @@
+
+For example, if a plugin is dependent _only_ on one or more of the modules in the table above, and declares the module dependencies in `plugin.xml`, then any product based on the IntelliJ Platform will load it.
+
+### Modules Specific to Functionality
+More specialized functionality is also delivered via modules in IntelliJ Platform-based products. 
+For example, the `com.intellij.modules.java` module supports the language-specific Java functionality. 
+If a plugin uses functionality from this module, such as Java-specific inspections and refactoring, it must declare a dependency on this module.
+
+Note that not all products define and declare modules. 
+For example, PhpStorm does not have its own module, but the product itself depends on (and ships with) the PHP language plugin. 
+A plugin project is compatible with PHP functionality if it declares a dependency on this PHP language plugin.
+
+The following table lists modules or built-in plugins that provide specific functionality, and the products that currently ship with them. 
+Module FQN is listed on the left, an overview of functionality is listed in the middle column, and availability in products is listed on the right:
+
+| Plugin Dependency:<br>Module or Built-in Plugin  | Functionality  | IntelliJ Platform-Based Products  | 
+|--------------------------------------------------|----------------|-----------------------------------|
+| `com.intellij.modules.java`          | **Java** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework  | IntelliJ IDEA, Android Studio         |
+| `com.intellij.modules.ultimate`      | JavaScript and TypeScript language PSI Models, Java EE, JVM Frameworks: Spring, Play, Grails...  | IntelliJ IDEA Ultimate Edition      |
+| `com.intellij.modules.androidstudio` | Android SDK Platform, Build Tools, Platform Tools, SDK Tools | Android Studio                                                          |
+| `com.intellij.modules.appcode`       | CocoaPods, Core Data Objects, Device & Simulator Support  | AppCode                                                                    |
+| `com.intellij.modules.cidr.lang`     | **C, C++, Objective-C/C++** language PSI Model, Swift/Objective-C Interaction, Inspections, Intentions, Completion, Refactoring, Test Framework  | AppCode, CLion |
+| `com.intellij.modules.cidr.debugger` | Debugger Watches, Evaluations, Breakpoints, Inline Debugging  | AppCode, CLion, RubyMine                                               |
+| `com.intellij.modules.clion`         | CMake, Profiler, Embedded Development, Remote Development, Remote Debug, Disassembly | CLion                                           |
+| `com.intellij.modules.database`      | **SQL** language PSI Model, Inspections, Completion, Refactoring, Queries | IntelliJ IDEA Ultimate Edition, DataGrip, GoLand, PhpStorm, PyCharm, Rider, RubyMine |
```

Comment:
`com.intellij.modules.database` is included into [Database Tools and SQL](https://plugins.jetbrains.com/plugin/10925-database-tools-and-sql) plugin which is currently also bundled with WebStorm, CLion and AppCode. Also note that it's available in Professional edition of PyCharm only, not in Community Edition.

Reply:
I have updated the table to list `com.intellij.database`, and the application information from the [DataGrip Features Page](https://www.jetbrains.com/datagrip/features/).